test/spec/mfa-challenge.js

```diff
@@ -1291,6 +1291,53 @@ describe('MFA_CHALLENGE', function () {
       }
     });
 
+    util.itMakesCorrectRequestResponse({
+      title: 'allows polling for push with transactionCallBack',
+      setup: {
+        status: 'mfa-challenge-push',
+        calls: [
+          {
+            request: {
+              uri: '/api/v1/authn/factors/opf492vmb3s1blLTs0h7/verify',
+              data: {
+                stateToken: '00T4jcVNRzJy5dkWJ4P7c9051dY3FUYY9O2zvbU_vI'
+              }
+            },
+            response: 'mfa-challenge-push'
+          },
+          {
+            request: {
+              uri: '/api/v1/authn/factors/opf492vmb3s1blLTs0h7/verify',
+              data: {
+                stateToken: '00T4jcVNRzJy5dkWJ4P7c9051dY3FUYY9O2zvbU_vI'
+              }
+            },
+            response: 'mfa-challenge-push'
+          },
+          {
+            request: {
+              uri: '/api/v1/authn/factors/opf492vmb3s1blLTs0h7/verify',
+              data: {
+                stateToken: '00T4jcVNRzJy5dkWJ4P7c9051dY3FUYY9O2zvbU_vI'
+              }
+            },
+            response: 'success'
+          }
+        ]
+      },
+      execute: function (test) {
+        test.transactionCallbackFn = jasmine.createSpy('spy');
+        return test.trans.poll({
+          delay: 0, 
+          transactionCallBack: test.transactionCallbackFn
+        });
+      },
+      expectations: function (test) {
+        expect(test.transactionCallbackFn.calls.count()).toBe(2);
+        expect(test.transactionCallbackFn.calls.argsFor(0)[0]).toMatchSnapshot();
```

Comment:
I have not yet grokked `util.itMakesCorrectRequestResponse` yet, but why test for two calls and then only match the snapshot for one?

Reply:
the responses  that we get during the poll is same in this case as this is just polling.